server/controllers/contest.js

```diff
@@ -0,0 +1,95 @@
+const asyncHandler = require("express-async-handler");
+const Contest = require('../models/Contest')
+
+// handler for creating a new contest object
+exports.createContest = asyncHandler(async (req, res, next) => {
+    const { title, description, user, prizeAmount, deadline } = req.body;
+
+    const contest = await Contest.create({ title, description, prizeAmount, deadline });
+
+    if (contest) {
+        res.status(201).json({
+            success: {
+                contest: {
+                    created: contest.created,
+                    title: contest.title,
+                    description: contest.description
+                }
+            }
+        });
+    } else {
+        res.status(400);
+        throw new Error("invalid contest data");
+    }
+
+    next()
+
+});
+
+// handler for updating a contest object
+exports.updateContest = asyncHandler(async (req, res, next) => {
+
+    try {
+        const id = req.params.id
+        const title = req.body.title
+        const description = req.body.description;
+        const prizeAmount = req.body.prizeAmount;
+
+        const updated = await Contest.updateOne({ id: id }, { title: req.body.title, description: description, prizeAmount: prizeAmount })
+
+        if (updated.nModified > 0) {
+            res.status(200).json({
+                success: {
+                    contest: {
+                        title: title,
+                        description: description,
+                        prizeAmount: prizeAmount,
+                    }
+                }
+            });
+        }
+
+
+    } catch (error) {
+        res.status(400)
```

Comment:
Please update this with a status code of 500, as it is a server error.

Reply:
noted. Thank you.